packages/manager/src/views/WebPropertyDetailPage/components/Ephemeral/Ephemeral.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Caption,
+  Tbody,
+  Td,
+  ExpandableRowContent
+} from '@patternfly/react-table';
+import {
+  Bullseye,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Label,
+  Title
+} from '@patternfly/react-core';
+import {
+  AngleDownIcon,
+  CodeBranchIcon,
+  ExternalLinkAltIcon,
+  OutlinedClockIcon,
+  SearchIcon
+} from '@patternfly/react-icons';
+import { TEphemeralEnv } from '@app/services/ephemeral/types';
+import { useFormatDate } from '@app/hooks';
+import { useState } from 'react';
+
+type Props = {
+  ephemeralEnvs: TEphemeralEnv[] | undefined;
+  isSuccess: boolean;
+};
+
+export const Ephemeral = ({ isSuccess, ephemeralEnvs }: Props): JSX.Element => {
+  const isEnvEmpty = ephemeralEnvs?.length === 0;
+  const URL_LENGTH_LIMIT = 40;
+  const [isRowExpanded, setIsRowExpanded] = useState<Record<string, boolean>>({});
+  const formatDate = useFormatDate();
+  const onToggleRowExpanded = (id: string) => {
+    const state = { ...isRowExpanded };
+    if (state?.[id]) {
+      state[id] = !state[id];
+    } else {
+      state[id] = true;
+    }
+    setIsRowExpanded(state);
+  };
+  const msToTime = (ms: number) => {
+    const minutes = Number((ms / (1000 * 60)).toFixed(0));
+    const hours = Number((ms / (1000 * 60 * 60)).toFixed(0));
+    const days = Number((ms / (1000 * 60 * 60 * 24)).toFixed(0));
+    if (minutes < 60) return `${minutes} min`;
+    if (hours < 24) return `${hours} hr`;
+    return `${days} Days`;
+  };
+  const getExpiresTime = (createdAt: string, totalTime: number) => {
```

Comment:
Keep this outside the component as its pure function. 

This will causes re-render

Reply:
Also why manually doing this. dayjs can create this naa